### ISSUE TITLE
Simplify styles definition

### ISSUE THREAD

**Issue by woothu** [atomiks/tippyjs#720]
## Problem

Styles of tooltip could be easier to define.

## Solution

I think that the '.tippy-box' element could be responsible for styling tooltip. 
I don't have insights into style nuances, but I think it should be manageable to set styling based on this one element parameters.

I have made a demo fiddle which partially shows what I mean by simplifying style definition:
http://jsfiddle.net/zu7ewLsx/19/

One thing I didn't get on fiddle is proper arrow background-color. I wanted to get the following result, which demand ::before element to inherit background-color and ::after element to inherit border-color

```css
.tippy-box[data-theme~='ugly-example'] {
  border-color: green;
  text-color: black;
  background-color: dodgerblue;
}
```
Result: 
![image](https://user-images.githubusercontent.com/32914662/75791947-9a506580-5d6d-11ea-8dd3-be0325c65290.png)

Advantages:
- super easy theme API,
- super light themes,
- if 'styleClasses' prop was added to API (string with classes passed to tippy-box), user can define styles by existing classes, which for now is not possible i.e 
```js
{
  styleClasses: 'user-text-color-class user-background-color-class user-border-class'
}
```

Disadvantages:
- might be breaking change (Although it is possible that current selectors are more specific and no breaking change would occur)
- SVG arrows might need JS handling to also have this behavior

Overall I think that the idea of manipulating the whole design of tooltip just by these three parameters is simplifying the API greatly :)

**Comment by atomiks** [comment]
A PR implementing this would be great assuming it's fully backwards-compat.

**Comment by woothu** [comment]
Do you mean backwards compatibility with docs examples or any possible styles?

**Comment by atomiks** [comment]
I mean if a user updates the version, and their styles break, it's not backwards-compat

**Comment by woothu** [comment]
It is possible to only cover some possible users theme usages making any change on CSS

**Comment by woothu** [comment]
solved with #727
